internal/controller/api_to_config.go

```diff
@@ -522,10 +522,13 @@ func localPrefPrefixesToMap(withLocalPref []v1beta1.LocalPrefPrefixes) (localPre
 				lpMap = res.localPrefForPrefixV6
 			}
 
-			_, ok := lpMap[p]
-			if ok {
+			v, ok := lpMap[p]
+			if ok && v != pfxs.LocalPref {
 				return localPrefPrefixes{}, fmt.Errorf("multiple local prefs specified for prefix %s", p)
 			}
+			if ok && v == pfxs.LocalPref {
```

Comment:
I wonder if we should even error here. What happens if we add the same local pref twice (either in the same frrconfiguration or when merging)?

Reply:
>  if we add the same local pref twice in the same frrconfiguration
what frrconfig  do you have in mind? I do not see the use case to be honest.

> if we add the same local pref twice when merging
This sounds very legit, then we should not error we should just let it pass and ignore duplicates.
Do you have an example of that use case, is unclear why two frrconfig (two actors) would like to use the same prefix/local pref twice?

Reply:
> > if we add the same local pref twice in the same frrconfiguration
> 
> what frrconfig do you have in mind? I do not see the use case to be honest.
> 

The end result I expect is having them squashed.

> > if we add the same local pref twice when merging
> 
> This sounds very legit, then we should not error we should just let it pass and ignore duplicates. Do you have an example of that use case, is unclear why two frrconfig (two actors) would like to use the same prefix/local pref twice?

Because they agree the localpref must be that particular value I guess.

Reply:
(I think) this function does not know about merges, it works per neighbor. wdyt about enforcing that a single `[]LocalPrefPrefixes` cannot have the same localPref defined multiple times + the same prefix cannot appear in multiple `[]Prefixes` (of the same []LocalPrefPrefixes)?

Reply:
@oribon  Can you elaborate a bit more? What would be the error when ` the same prefix cannot appear in multiple []Prefixes (of the same []LocalPrefPrefixes)?`  Should we handle different the following case?
Like error in the first, do not error in the second?

```
PrefixesWithLocalPref: []v1beta1.LocalPrefPrefixes{
{
Prefixes:  []string{"192.0.2.0/24", "192.0.2.0/24"},LocalPref: 100}
}
```

```
PrefixesWithLocalPref: []v1beta1.LocalPrefPrefixes{
{
Prefixes:  []string{"192.0.2.0/24"},LocalPref: 100},
Prefixes:  []string{"192.0.2.0/24"},LocalPref: 100}
},
```

Reply:
I'm suggesting to error on both cases (to enforce FRRConfigurations to be cleaner).